### ISSUE TITLE
Display problems in (java?) desktop GUI applications

### ISSUE THREAD

**Issue by mbtaylor** [opencadc/science-platform#578]
I have experienced issues with the display of java GUI applications in a science platform desktop session.

The issues are intermittent, have happened over a long period, and are in some cases just cosmetic and in other cases more serious.

Miscellaneous observations:
* I've seen problems in topcat and Aladin. I haven't seen problems in ds9, which leads me to believe this might be a java issue.  I haven't really tried other desktop applications.
* One cosmetic effect is scrambling of toolbar button icons, see e.g. the screenshot [here](https://github.com/opencadc/science-containers-old/pull/19#issuecomment-1824923026) (posted by @severingaudet)
* A more serious effect is in plot windows in topcat - they display OK when first plotted, but subsequent user interaction (pan/zoom) doesn't work properly, e.g. moves the axes around but not the plotted points
* In Aladin I've had all-sky imagery claim in the GUI that it is loaded, but not visible in the display
* I noticed these issues first in 06/2021, and again now (11/2023).  I haven't really played with the science platform in between.
* Problems are intermittent - e.g. at the moment it's behaving nicely.  I'm not sure, but I have the impression that problems are more common in Canadian daytime than the middle of the Canadian night, so this could be related to loading?
* I had an idea in 2021 that fiddling with [java2d system properties](https://docs.oracle.com/javase/8/docs/technotes/guides/2d/flags.html), e.g. `-Dsun.java2d.xrender=false` might fix it.  But as long as it's not misbehaving consistently I can't test that.

**Comment by mbtaylor** [comment]
Here's a screenshot of a topcat plot misbehaving.
![shot1](https://github.com/opencadc/science-platform/assets/2581893/58fa6c94-da58-45a7-a139-cebeaf1151ea)
It's not very revealing, but take my word for it there should be a sky full of points visible.
I just ran `topcat :skysim:1000000` from a topcat-terminal and hit the Sky Plot toolbar button.  Plot comes up nicely, but when I try to drag it around using the mouse, the axes move but the plotted points go away.  Note also a couple of missing toolbar icons and scrambled pixelmaps on a horizontal and a vertical divider in the plot window - those are not really a problem, but tend to accompany the plot problems.

**Comment by brianmajor** [comment]
I don’t think this is the issue (is OpenGL used?) but posting for the record : https://bbs.archlinux.org/viewtopic.php?id=269181

**Comment by mbtaylor** [comment]
I'm not using OpenGL explicitly; I haven't attempted to check whether Java uses it but I would think not (there's no 3d graphics going on here).  And the symptoms don't look very similar.  So probably that's not the same issue.

**Comment by mbtaylor** [comment]
And here's a shot of Aladin misbehaving:
![aladin](https://github.com/opencadc/science-platform/assets/2581893/91a117a8-803c-4849-be65-f11a5af8202a)
Icons on the left and right are scrambled, and the sky display is blank.  Interacting with the plot appears to work in that the zoom slider bottom right moves, but the sky display does not respond.

**Comment by brianmajor** [comment]
Another post that may provide insight:  https://stackoverflow.com/questions/35852534/java-swing-applications-not-rendering-properly-with-vnc

**Comment by brianmajor** [comment]
Raising the priority on this as it is clear it is affecting all Java-based desktop-apps.

**Comment by brianmajor** [comment]
@ijiraq - You mentioned that you have an idea of what the issue may be.  Could you please post the details so we can give it a try?
